samples/encode-decode-worker/js/stream_worker.js

```diff
@@ -0,0 +1,323 @@
+     return new TransformStream({
+       start(controller) {
+         this.decoder = decoder = new VideoDecoder({
+           output: frame => controller.enqueue(frame),
+           error: (e) => {
+              self.postMessage({severity: 'fatal', text: `Init Decoder error: ${e.message}`});
+           }
+         });
+       },
+       transform(chunk, controller) {
+         if (this.decoder.state != "closed") {
+           if (chunk.type == "config") {
+              let config = JSON.parse(chunk.config);
+              VideoDecoder.isConfigSupported(config).then((decoderSupport) => {
+                if(decoderSupport.supported) {
+                  this.decoder.configure(decoderSupport.config);
+                  self.postMessage({text: 'Decoder successfully configured:\n' + JSON.stringify(decoderSupport.config)});
+                 // self.postMessage({text: 'Decoder state: ' + JSON.stringify(this.decoder.state)});
+                } else {
+                self.postMessage({severity: 'fatal', text: 'Config not supported:\n' + JSON.stringify(decoderSupport.config)});
+                }
+              })
+              .catch((e) => {
+                 self.postMessage({severity: 'fatal', text: `Configuration error: ${e.message}`});
+              })
+           } else {
+             try {
+              // self.postMessage({text: 'size: ' + chunk.byteLength + ' seq: ' + chunk.seqNo + ' kf: ' + chunk.keyframeIndex + ' delta: ' + chunk.deltaframeIndex + ' dur: ' + chunk.duration + ' ts: ' + chunk.timestamp + ' ssrc: ' + chunk.ssrc + ' pt: ' + chunk.pt + ' tid: ' + chunk.temporalLayerId + ' type: ' + chunk.type});
+               const queue = this.decoder.decodeQueueSize;
+               decqueue_update(queue);
+               const before = performance.now();
+               this.decoder.decode(chunk);
+               const after = performance.now();
+               const duration = after - before;
+               dec_update(duration);
+             } catch (e) {
+               self.postMessage({severity: 'fatal', text: 'Derror size: ' + chunk.byteLength + ' seq: ' + chunk.seqNo + ' kf: ' + chunk.keyframeIndex + ' delta: ' + chunk.deltaframeIndex + ' dur: ' + chunk.duration + ' ts: ' + chunk.timestamp + ' ssrc: ' + chunk.ssrc + ' pt: ' + chunk.pt + ' tid: ' + chunk.temporalLayerId + ' type: ' + chunk.type});
+               self.postMessage({severity: 'fatal', text: `Catch Decode error: ${e.message}`});
+             }
+           }
+         }
+       }
+     });
+   }
+
+   EncodeVideoStream(self, config) {
+     return new TransformStream({
+       start(controller) {
+         this.frameCounter = 0;
+         this.seqNo = 0;
+         this.keyframeIndex = 0;
+         this.deltaframeIndex = 0;
+         this.pending_outputs = 0;
+         this.encoder = encoder = new VideoEncoder({
+           output: (chunk, cfg) => {
+             if (cfg.decoderConfig) {
+               // self.postMessage({text: 'Decoder reconfig!'});
```

Comment:
Ditto.

Reply:
This isn't particularly useful even as a debug message, so I've removed it.